moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -288,11 +295,24 @@ bool PlanningPipeline::generatePlan(const planning_scene::PlanningSceneConstPtr&
                      req_adapter->getDescription().c_str(), status.message.c_str());
         break;
       }
+
+      // check for timeout
+      if (std::chrono::duration<double>(clock::now() - plan_start_time).count() >= allowed_planning_time)
+      {
+        throw timeout_error;
```

Comment:
I don't think that a timeout is a reason to throw. Instead, you should return an unsuccessful MotionPlanResponse with the [timeout error code](https://github.com/ros-planning/moveit_msgs/blob/9756797dcbc293c7cae66541a7bc0c7acec85892/msg/MoveItErrorCodes.msg#L12).
For example:
```cpp
        RCLCPP_ERROR(node_->get_logger(),
                     "Planning pipeline couldn't solve the motion planning problem within the allowed planning time (%lf s). Last element called '%s'",req_adapter->getDescription().c_str()) ;
        res.error_code = moveit::core::MoveItErrorCode::TIMED_OUT;
        break;
```